modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Fields.tsx

```diff
@@ -383,6 +394,76 @@ const EditFDSFieldModalContent = ({
 	const fdsFieldLabelInputId = `${namespace}fdsFieldLabelInput`;
 	const fdsFieldRendererSelectId = `${namespace}fdsFieldRendererSelectId`;
 
+	const options = FDS_INTERNAL_CELL_RENDERERS.map(
+		(renderer: FDSInternalCellRenderer) => ({
+			label: renderer.label!,
+			value: renderer.name!,
+		})
+	);
+
+	options.push(
+		...cellRendererCETs.map((item) => ({
+			label: item.name,
+			value: item.erc,
+		}))
+	);
+
+	const CellRendererDropdown = ({
+		cellRenderers,
+		namespace,
+		onItemClick,
+	}: {
+		cellRenderers: {
+			label: string;
+			value: string;
+		}[];
+		namespace: string;
+		onItemClick: Function;
+	}) => {
+		const cellRendererERCs = cellRendererCETs.map(
+			(cellRendererCET) => cellRendererCET.erc
+		);
+
+		return (
+			<ClayDropDown
+				menuElementAttrs={{
+					className: 'fds-cell-renderers-dropdown-menu',
+				}}
+				trigger={
+					<ClayButton
+						aria-labelledby={`${namespace}cellRenderersLabel`}
+						className="form-control form-control-select form-control-select-secondary"
+						displayType="secondary"
+						id={fdsFieldRendererSelectId}
+					>
+						{selectedFDSFieldRenderer
+							? selectedFDSFieldRenderer
+							: Liferay.Language.get('choose-an-option')}
+					</ClayButton>
+				}
+			>
+				<ClayDropDown.ItemList items={cellRenderers} role="listbox">
+					{cellRenderers.map((cellRenderer) => (
+						<ClayDropDown.Item
+							className="align-items-center d-flex justify-content-between"
+							key={cellRenderer.value}
+							onClick={() => onItemClick(cellRenderer.label)}
+							roleItem="option"
+						>
+							{cellRenderer.label}
+
+							{cellRendererERCs.includes(cellRenderer.value) && (
```

Comment:
I believe we can clarify and make code more robust here by setting `options` / `cellRenderers`  prop to be Array<[CellRenderer](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/utils/renderer.ts#L60)> type.

With that, the check would be 
```ts
{cellRenderer.type === 'clientExtension' && {
   ....
}
```

Reply:
What would be used as the label? Currently we use `{cellRenderer.label}` but `label` isn't a property on `CellRenderer`.